dGame/dComponents/InventoryComponent.cpp

```diff
@@ -989,6 +898,84 @@ void InventoryComponent::UnEquipItem(Item* item) {
 	}
 }
 
+void InventoryComponent::HandlePossession(Item* item) {
+	auto* characterComponent = m_Parent->GetComponent<CharacterComponent>();
+	if (!characterComponent) return;
+
+	auto* possessorComponent = m_Parent->GetComponent<PossessorComponent>();
+	if (!possessorComponent) return;
+
+	// don't do anything if we are busy dismounting
+	if (possessorComponent->GetIsDismounting()) return;
+
+	// check to see if we are already mounting something
+	auto* currentlyPossessedEntity = EntityManager::Instance()->GetEntity(possessorComponent->GetPossessable());
+	auto currentlyPossessedItem = possessorComponent->GetMountItemID();
+
+	if (currentlyPossessedItem) {
+		if (currentlyPossessedEntity) possessorComponent->Dismount(currentlyPossessedEntity);
+		return;
+	}
+	// check again cause sometimes we need to
+	if (possessorComponent->GetIsDismounting()) return;
```

Comment:
Due to the nature of possession being a handshake between server and client, somethings can be done in a weird order. I think this was necessary to prevent some issues, but I'll double check